alerting/src/main/kotlin/org/opensearch/alerting/model/DocumentLevelTriggerRunResult.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.model
+
+import org.opensearch.alerting.alerts.AlertError
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.xcontent.ToXContent
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.script.ScriptException
+import java.io.IOException
+import java.time.Instant
+
+data class DocumentLevelTriggerRunResult(
+    override var triggerName: String,
+    var triggeredDocs: List<String>,
+    override var error: Exception?,
+    var actionResults: MutableMap<String, ActionRunResult> = mutableMapOf()
+) : TriggerRunResult(triggerName, error) {
+
+    @Throws(IOException::class)
+    @Suppress("UNCHECKED_CAST")
```

Comment:
can we avoid this?

Reply:
It is due to `actionResults` has to be casted as `MutableMap<String, ActionRunResult>` from a generic map in this function. This is an issue for Query Level and Bucket level